src/main.rs

```diff
@@ -42,6 +44,174 @@ impl Shell {
             history: VecDeque::new(),
         }
     }
+
+    pub fn print_prompt(&self) {
+        let prompt_prefix = self.modes.iter().rev().fold(String::new(), |acc, mode| {
+            acc +
+            if mode.value {
+                "+ "
+            } else {
+                "- "
+            }
+        });
+        print!("{}", prompt_prefix);
+
+        let cwd = env::current_dir().ok().map_or("?".to_string(),
+                                                 |ref p| p.to_str().unwrap_or("?").to_string());
+
+        print!("ion:{}# ", cwd);
+        if let Err(message) = stdout().flush() {
+            println!("{}: failed to flush prompt to stdout", message);
+        }
+    }
+
+    fn on_command(&mut self, command_string: &str, commands: &HashMap<&str, Command>) {
+        let max_history: usize = 1000; // TODO temporary, make this configurable
```

Comment:
Well, why not fetch it from a variable, and then make it configurable by setting that var in the rc file?